### PR TITLE
Search: support URL query params

### LINKED ISSUE #24379: Support search query parameters in dashboard search page
**What would you like to be added**:

I'd like to be able to link to a pre-filled [dashboard search page](https://grafana.com/docs/grafana/latest/reference/search/) using query parameters similar to the [dashboard search api](https://grafana.com/docs/grafana/latest/http_api/folder_dashboard_search/).

e.g. https://my-grafana/dashboards?tag=service&query=test would show me this page:

Screen Shot 2020-05-07 at 11 12 52

**Why is this needed**:

This would allow me to bookmark and share searches.

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:
Save search filters i.e. tags, sort, starred and search query to URL params. This would enable sharing the URLs for filtered search results. 

**Which issue(s) this PR fixes**:

Fixes #24379

**Special notes for your reviewer**:
- The URL params are also enabled for Manage Dashboards view. Not sure if this is necessary and if the main search is opened from manage dashboards view (not sure if this happens often) the main search params will overwrite the ones from manage dashboards. 
- The main search's URL params are cleared when the search is closed. Maybe there'd be some action for manage dashboards view to clear all the filters (now only clearing tags is possible).

### CODE REVIEW

**Review by alexanderzobnin — CHANGES_REQUESTED** [review]
Looks good! The only thing I noticed is a folder param. When you open search from the folder (by clicking at folder name in the dashboard selector), `folder:current` is adding to search and `folder=current` - to the URL params. If you open URL with `folder=current` and `query` param is empty, `folder:current` will be added to the search input. But if both `folder=current` and `query` params are specified, `query` will be used as is. For instance:
- `?query=folder:current%20prometheus&search=open&folder=current` - input will be `folder:current prometheus` and search will be invoked in the current folder
- `?query=prometheus&search=open&folder=current` - input will be `prometheus` and search will be invoked globally.

So, probably, `folder` and `query` params should be syncronized against each other?

File: public/app/core/components/Select/SortPicker.tsx [hunk 1]
```diff
@@ -19,15 +20,17 @@ const getSortOptions = () => {
 };
 
 export const SortPicker: FC<Props> = ({ onChange, value, placeholder }) => {
-  return (
-    <AsyncSelect
+  // Using sync Select and manual options fetching here since we need to find the selected option by value
+  const { loading, value: options } = useAsync<SelectableValue[]>(getSortOptions, []);
+
+  return !loading ? (
```
[the remark this answers is not in the document]
Comment (Clarity-89) [reply public/app/core/components/Select/SortPicker.tsx:26]: `FC` type expects React component or `null` as a return value, using `&&` will return `false` instead. It works with our ts config, but results in error with `nullStrictChecks` on.

[the remark this answers is not in the document]
Comment (alexanderzobnin) [reply public/app/core/components/Select/SortPicker.tsx:26]: Aha, ok, got it!

**Review by alexanderzobnin — APPROVED** [review]
LGTM!

**Pull request merged**